safe-global/safe-wallet-web
# [Multichain] Feat: disable adding new chains to safes created from old mastercopy versions [SW-167]

## What it solves

Resolves: https://www.notion.so/safe-global/Do-not-offer-adding-networks-for-1-1-1-created-Safes-9608b9ee16a549e891b795fe18b5e831?pvs=4

## How this PR fixes it
- Gets the mastercopy/factory address used to create a given safe, and checks if there is a corresponding deployment within the list of supported versions.
- If there are no matching deployments, display a message informing the user that this safe cannot be replayed.

## How to test it
- Open a safe that was created as a 1.1.1 safe. 
- Click the network selector in the top right corner. It should show a message saying that the safe cannot be replayed.
- Click on the context menu of the safe in the sidebar or the accounts page. Click the button to add another network. It should show a message saying that the safe cannot be replayed.

## Screenshots
![image](https://github.com/user-attachments/assets/afb8684d-58d5-49c7-a6ef-ab0f173ba38a)

![image](https://github.com/user-attachments/assets/bd5a3f68-18bc-499e-888b-31135a65d330)

## Checklist
